### PR TITLE
Customer downloadable products GraphQL

### PR DESCRIPTION
**Pull request by ldusan84**
### Description (*)

Moved customer downloadable products into a new module and added tests.

### Fixed Issues (if relevant)

1. magento/graphql-ce#685: Extract logic related to customer from DownloadableGraphQl module into new one

### Manual testing scenarios (*)

1. Get token as customer:

```
mutation {
  generateCustomerToken(
    email: "customer@example.com"
    password: "password"
  ) {
    token
  }
}
```
2. Get customer's downloadable products:
```
{
    customerDownloadableProducts{
        items{
            order_increment_id
            date
            status
            download_url
            remaining_downloads
        }
    }
}
```

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/CustomerDownloadableGraphQl/composer.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+    "name": "magento/module-customer-downloadable-graph-ql",
+    "description": "N/A",
+    "type": "magento2-module",
+    "require": {
+        "php": "~7.1.3||~7.2.0",
```
Comment (lenaorobei) [inline app/code/Magento/CustomerDownloadableGraphQl/composer.json]:
``` suggestion
        "php": "~7.1.3||~7.2.0||~7.3.0",
```

File: app/code/Magento/CustomerDownloadableGraphQl/etc/schema.graphqls [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    items: [CustomerDownloadableProduct] @doc(description: "List of purchased downloadable items")
+}
+
+type CustomerDownloadableProduct {
+    order_increment_id: String
+    date: String
+    status: String
+    download_url: String
+    remaining_downloads: String
+}
```
Comment (lenaorobei) [inline app/code/Magento/CustomerDownloadableGraphQl/etc/schema.graphqls]:
``` suggestion
}

```

File: dev/tests/api-functional/testsuite/Magento/GraphQl/CustomerDownloadableProduct/CustomerDownloadableProductTest.php [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+class CustomerDownloadableProductTest extends GraphQlAbstract
+{
+
+    /**
+     * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     * @magentoApiDataFixture Magento/Downloadable/_files/product_downloadable.php
+     * @magentoApiDataFixture Magento/Downloadable/_files/order_with_downloadable_product.php
+     */
+    public function testGetCustomerDownloadableProducts()
```
Comment (lenaorobei) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/CustomerDownloadableProduct/CustomerDownloadableProductTest.php]: This test can be simplified. Please have a look at `\Magento\GraphQl\Downloadable\CustomerDownloadableProductsTest::testCustomerDownloadableProducts`.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please address my comments and remove duplicated tests `\Magento\GraphQl\Downloadable\CustomerDownloadableProductsTest`.

**Comment by ldusan84** [comment]
@lenaorobei I changed the things that you mentioned, please check.

**Comment by lenaorobei** [comment]
@ldusan84 could you please pull latest mainline changes. We delivered recently https://github.com/magento/graphql-ce/pull/412 and it will impact your PR.

**Comment by ldusan84** [comment]
@lenaorobei that's done.

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5718](https://jira.corp.magento.com/browse/ENGCOM-5718) has been created to process this Pull Request
:eight_spoked_asterisk: @lenaorobei, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by lenaorobei** [comment]
@magento run Functional Tests build

**Pull request merged**